src/Operators/arrayinterpolate.jl

```diff
@@ -0,0 +1,165 @@
+function interpolate_slab(
+    field::Fields.Field,
+    slabidx::Fields.SlabIndex,
+    (I1,)::Tuple{<:AbstractArray},
+)
+    space = axes(field)
+    x = zero(eltype(field))
+    QS = Spaces.quadrature_style(space)
+    Nq = Spaces.Quadratures.degrees_of_freedom(QS)
+
+    for i in 1:Nq
+        ij = CartesianIndex((i,))
+        x += I1[i] * Operators.get_node(space, field, ij, slabidx)
+    end
+    return x
+end
+
+function interpolate_slab(
+    field::Fields.Field,
+    slabidx::Fields.SlabIndex,
+    (I1, I2)::Tuple{<:AbstractArray, <:AbstractArray},
+)
+    space = axes(field)
+    x = zero(eltype(field))
+    QS = Spaces.quadrature_style(space)
+    Nq = Spaces.Quadratures.degrees_of_freedom(QS)
+
+    for j in 1:Nq, i in 1:Nq
+        ij = CartesianIndex((i, j))
+        x += I1[i] * I2[j] * Operators.get_node(space, field, ij, slabidx)
+    end
+    return x
+end
+
+function interpolate_slab_level(
+    field::Fields.Field,
+    h::Integer,
+    Is::Tuple,
+    zcoord,
+)
+    space = axes(field)
+    vert_topology = Spaces.vertical_topology(space)
+    vert_mesh = vert_topology.mesh
+    Nz = Spaces.nlevels(space)
+
+    velem = Meshes.containing_element(vert_mesh, zcoord)
+    ξ3, = Meshes.reference_coordinates(vert_mesh, velem, zcoord)
+    if space isa Spaces.FaceExtrudedFiniteDifferenceSpace
+        v_lo = velem - half
+        v_hi = velem + half
+    elseif space isa Spaces.CenterExtrudedFiniteDifferenceSpace
+        if ξ3 < 0
+            if Topologies.isperiodic(Spaces.vertical_topology(space))
+                v_lo = mod1(velem - 1, Nz)
+            else
+                v_lo = max(velem - 1, 1)
+            end
+            v_hi = velem
+            ξ3 = ξ3 + 1
+        else
+            v_lo = velem
+            if Topologies.isperiodic(Spaces.vertical_topology(space))
+                v_hi = mod1(velem + 1, Nz)
+            else
+                v_hi = min(velem + 1, Nz)
+            end
+            ξ3 = ξ3 - 1
+        end
+    end
+    f_lo = interpolate_slab(field, Fields.SlabIndex(v_lo, h), Is)
+    f_hi = interpolate_slab(field, Fields.SlabIndex(v_hi, h), Is)
+    return ((1 - ξ3) * f_lo + (1 + ξ3) * f_hi) / 2
+end
+
+"""
+    interpolate_array(field, xpts, ypts)
+    interpolate_array(field, xpts, ypts, zpts)
+
+Interpolate a field to a regular array using pointwise interpolation.
+
+This is primarily used for plotting and diagnostics.
+
+# Examples
+
+```julia
+longpts = range(Geometry.LongPoint(-180.0), Geometry.LongPoint(180.0), length = 21)
+latpts = range(Geometry.LatPoint(-80.0), Geometry.LatPoint(80.0), length = 21)
+zpts = range(Geometry.ZPoint(0.0), Geometry.ZPoint(1000.0), length = 21)
+
+interpolate_array(field, longpts, latpts, zpts)
+```
+
+!!! note
+    Hypsography is not currently handled correctly.
+"""
+interpolate_array
```

Comment:
```suggestion
function interpolate_array end
```

I'd prefer we declare this a function for clarity